src/domain/ecoverse/ecoverse.module.ts

```diff
@@ -9,26 +9,38 @@ import { TypeOrmModule } from '@nestjs/typeorm';
 import { Ecoverse } from './ecoverse.entity';
 import { EcoverseResolverQueries } from './ecoverse.resolver.queries';
 import { EcoverseResolverMutations } from './ecoverse.resolver.mutations';
+import { ContextService } from '../context/context.service';
+import { ContextModule } from '../context/context.module';
 import { TagsetModule } from '../tagset/tagset.module';
 import { TagsetService } from '../tagset/tagset.service';
+import { ReferenceModule } from '../reference/reference.module';
+import { ReferenceService } from '../reference/reference.service';
+import { ChallengeModule } from '../challenge/challenge.module';
+import { ChallengeService } from '../challenge/challenge.service';
 
 @Module({
-  providers: [
-    EcoverseService,
-    UserGroupService,
-    EcoverseResolverQueries,
-    EcoverseResolverMutations,
-    AzureADStrategy,
-    TagsetService
-  ],
   imports: [
-    UserGroupModule,
+    ChallengeModule,
+    ContextModule,
     AuthenticationModule,
+    ReferenceModule,
+    TagsetModule,
     UserGroupModule,
     UserModule,
     TagsetModule,
     TypeOrmModule.forFeature([Ecoverse]),
   ],
+  providers: [
+    ChallengeService,
```

Comment:
A bunch of providers here are not used. UserGroupService, ContextService and Tagset service are injected in the Ecoverse service. Why are the rest added?